rules/no-array-for-each.js

```diff
@@ -314,14 +319,7 @@ function isFixable(callExpression, {scope, functionInfo, allIdentifiers, context
 	}
 
 	// Check `CallExpression.parent`
-	if (callExpression.parent.type !== 'ExpressionStatement') {
-		return false;
-	}
-
-	// Check `CallExpression.callee`
-	// Because of `ChainExpression` wrapper, `foo?.forEach()` is already failed on previous check keep this just for safety
-	/* c8 ignore next 3 */
-	if (callExpression.callee.optional) {
+	if (callExpression.parent.type !== 'ExpressionStatement' && callExpression.parent.type !== 'ChainExpression') {
```

Comment:
This is not right `() => foo?.forEach()` and many other cases.

Reply:
Should be good now, since I added additional check in https://github.com/sindresorhus/eslint-plugin-unicorn/pull/1753/files#diff-0a69d7bd0e300d6d5043f953d861ec9c922c3bdda5bd91a4b5e0040834fd1392R327